packages/synthetics-sdk-broken-links/src/link_utils.ts

```diff
@@ -132,15 +134,60 @@ export function setDefaultOptions(
     per_link_options: {},
   };
 
-  const objKeys = Object.keys(default_options) as Array<
+  const outputOptions: BrokenLinksResultV1_BrokenLinkCheckerOptions =
+    {} as BrokenLinksResultV1_BrokenLinkCheckerOptions;
+
+  const optionsKeys = Object.keys(defaulOptions) as Array<
     keyof BrokenLinksResultV1_BrokenLinkCheckerOptions
   >;
-  for (const key of objKeys) {
-    if (!(key in options) && key !== 'origin_url') {
-      /* eslint-disable */
-      (options as any)[key] = default_options[key];
+  for (const optionName of optionsKeys) {
+    // per_link_options is handled below
+    if (!(optionName in inputOptions) || optionName === 'per_link_options') {
+      // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+      (outputOptions as any)[optionName] = defaulOptions[optionName];
+    } else {
+      // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+      (outputOptions as any)[optionName] = inputOptions[optionName];
     }
   }
+
+  // Convert `input_options.per_link_options`, type: {[key: string]: PerLinkOption}
+  // to `output_options.per_links_options`, type: {[key: string]: BrokenLinksResultV1_BrokenLinkCheckerOptions_PerLinkOption}
+  const perLinkOptions: {
+    [key: string]: BrokenLinksResultV1_BrokenLinkCheckerOptions_PerLinkOption;
+  } = {};
+  for (const [url, perLinkOption] of Object.entries(
+    inputOptions.per_link_options || {}
+  )) {
+    let expected_status_code;
+    if (perLinkOption.expected_status_code !== undefined) {
+      if (typeof perLinkOption.expected_status_code === 'number') {
+        expected_status_code = {
+          status_value: perLinkOption.expected_status_code,
+        };
+      } else {
+        expected_status_code = {
+          status_class:
+            ResponseStatusCode_StatusClass[perLinkOption.expected_status_code],
+        };
+      }
+    } else {
+      expected_status_code = {
+        status_class: ResponseStatusCode_StatusClass.STATUS_CLASS_2XX,
+      };
+    }
```

Comment:
162-178 feel pretty isolated from the logic if this method -- maybe pull it out into a helper method?